### PR TITLE
Allow rules string as a configuration alternative

### PR DESCRIPTION
**Pull request by filipesperandio**
```
pmd:
  enabled: true
  config:
    rules:
      - "java-basic"
      - "java-design"
```

### CODE REVIEW

File: fixtures/rules/config.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "enabled": true,
+  "channel": "beta",
+  "config": {
+    "rules": "java-basic,java-design"
```
Comment (toddmazierski) [inline fixtures/rules/config.json]: Is there a reason why this isn't an array (instead of a string with a comma-separated list)?

Comment (filipesperandio) [reply fixtures/rules/config.json]: Because I am stupid I guess... =/

Comment (toddmazierski) [reply fixtures/rules/config.json]: Ha, no way! 😛 

Just a suggestion to simplify parsing. 😄

Comment (filipesperandio) [reply fixtures/rules/config.json]: Actually we just forward the entire thing, I think that was the reason I didn't bother to split, but I makes more sense from a `yml` perspective to have it as an array.

Comment (filipesperandio) [reply fixtures/rules/config.json]: @toddmazierski Adjusted.

File: test/ConfigTest.groovy [hunk 2]
```diff
@@ -5,18 +5,30 @@ class ConfigTest {
...
+  @Test(expected = IllegalArgumentException.class)
+  public void doesNotAllowToMixRulesAndFile() {
+    def config = new Config([configFile: "/usr/src/app/fixtures/rules/config.mix.json", codeFolder: "/usr/src/app/fixtures/rules"])
+    config.ruleSet()
+  }
+
+  @Test
+  public void acceptRulesSimpleNames() {
+    def config = new Config([configFile: "/usr/src/app/fixtures/rules/config.json", codeFolder: "/usr/src/app/fixtures/rules"])
+    assertEquals "java-basic,java-design", config.ruleSet()
```
Comment (toddmazierski) [inline test/ConfigTest.groovy]: Should this assertion be adjusted?

Comment (filipesperandio) [reply test/ConfigTest.groovy]: That's the argument for the command line, should still be a single string.

Comment (toddmazierski) [reply test/ConfigTest.groovy]: Oh, my mistake.

Should `config.mix.json` still contain a comma-separated string, though?

Comment (filipesperandio) [reply test/ConfigTest.groovy]: Does not impact, but it should be adjusted. 👍

**Pull request merged**